benchmark/benchmain/main.go

```diff
@@ -128,6 +128,9 @@ func streamBenchmark(startTimer func(), stopTimer func(uint64), benchFeatures st
 
 func unconstrainedStreamBenchmark(benchFeatures stats.Features, warmuptime, benchtime time.Duration) (uint64, uint64) {
 	sender, recver, cleanup := makeFuncUnconstrainedStream(benchFeatures)
+	if benchFeatures.EnablePreloader {
```

Comment:
This will invoke `makeFuncUnconstrainedStream` unnecessarily if the preloader is enabled. Why don't we use this instead:
```go
	var sender, recver func(int)
	var cleanup func()
	if benchFeatures.EnablePreloader {
		sender, recver, cleanup = makeFuncUnconstrainedStreamPreloaded(benchFeatures)
	} else {
		sender, recver, cleanup = makeFuncUnconstrainedStream(benchFeatures)
	}
```